### DISCUSSION TITLE
How to debug unit test using nvim

### DISCUSSION THREAD

**Discussion by lillo42** [xunit/xunit#2562] [Question] [answered]
Hi folks,

I don't know if here is the right place to ask it

Do you know how can I debug an unit test on nvim?  Maybe a better question would be how unit test works(how vscode start unit test)?
I'm trying to run it using  nvim with netcoredbg, debug application is fine but I'm not find anyway debug unit test

**Comment by bradwilson** [comment, marked as answer]
VS Code (or more specifically, Omnisharp) runs unit tests by utilizing the VSTest adapters (that's what `xunit.runner.visualstudio` is). The code for launching is here: https://github.com/OmniSharp/omnisharp-vscode/blob/1d477d2e0495a9a7d76c7856dc4fe1a46343b7e1/src/features/dotnetTest.ts#L344-L379

I'm not sure that's going to be super helpful, since it relies upon VS Code to launch the debugger on its behalf (owing to the fact that the .NET Debugger that Omnisharp is using is the closed-source debugger from Microsoft). I honestly have no idea what's involved in getting netcoredbg to launch `dotnet test` within its debugger; it uses child processes for what it does, so the debugger would need to also automatically attach to child processes while debugging.
